clicache/integration-test2/SNITests.cs

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Xunit;
+using PdxTests;
+using System.Collections;
+using System.Collections.Generic;
+using Xunit.Abstractions;
+
+namespace Apache.Geode.Client.IntegrationTests
+{
+    [Trait("Category", "Integration")]
+    public class SNITests : TestBase, IDisposable
+    {
+        string currentWorkingDirectory;
+        Process dockerProcess;
+        private readonly Cache cache_;
+
+        public SNITests(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+            currentWorkingDirectory = Directory.GetCurrentDirectory();
+            var clientTruststore = Config.SslClientKeyPath + @"/truststore_sni.pem";
+
+
+
+            var cacheFactory = new CacheFactory();
+            cacheFactory.Set("log-level", "none");
+            cacheFactory.Set("log-file", "c:/temp/SNITest-csharp.log");
+            cacheFactory.Set("ssl-enabled", "true");
+            cacheFactory.Set("ssl-truststore", clientTruststore);
+
+            cache_ = cacheFactory.Create();
+
+            var dc = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker-compose.exe", "-f " + Config.SniConfigPath + "/docker-compose.yml" + " up -d");
+            dc.WaitForExit();
+
+            var d = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker.exe", "exec -t geode gfsh run --file=/geode/scripts/geode-starter.gfsh");
+            d.WaitForExit();
+        }
+
+        public void Dispose()
+        {
+
+            var dockerComposeProc = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker-compose.exe", "stop");
+            dockerComposeProc.WaitForExit();
+
+            var dockerProc = Process.Start(@"C:\Program Files\docker\docker\resources\bin\docker.exe", "container prune -f");
```

Comment:
There is a Docker C++ SDK we could use rather than forking an executable.

Reply:
not that we have found, unfortunately...

Reply:
https://github.com/lasote/docker_client

Reply:
thx! we'll take a look at this for future improvements, current deadline isn't going to allow

Reply:
Still, it is odd to hard code the docker executable paths.

Reply:
Oh good catch. I didn't even notice that initially over the whole, why not use the SDK thing. If the absolute path is necessary, which I doubt it is, then use CMake `find_program` and inject. It should be sufficient to just make sure docker is in the PATH or add it to the PATH in the CTest definition.